app/abci.go

```diff
@@ -11,48 +11,50 @@ import (
 )
 
 func (app *App) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) (res abci.ResponseBeginBlock) {
-	tracectx, topSpan := app.tracingInfo.Start("Block")
+	tracectx, topSpan := app.GetBaseApp().TracingInfo.Start("Block")
 	topSpan.SetAttributes(attribute.Int64("height", req.Header.Height))
-	app.tracingInfo.BlockSpan = &topSpan
-	app.tracingInfo.SetContext(tracectx)
-	_, beginBlockSpan := (*app.tracingInfo.Tracer).Start(app.tracingInfo.GetContext(), "BeginBlock")
+	app.GetBaseApp().TracingInfo.BlockSpan = &topSpan
+	app.GetBaseApp().TracingInfo.SetContext(tracectx)
+	_, beginBlockSpan := (*app.GetBaseApp().TracingInfo.Tracer).Start(app.GetBaseApp().TracingInfo.GetContext(), "BeginBlock")
 	defer beginBlockSpan.End()
 	return app.BaseApp.BeginBlock(ctx, req)
 }
 
 func (app *App) MidBlock(ctx sdk.Context, height int64) []abci.Event {
-	_, span := app.tracingInfo.Start("MidBlock")
+	_, span := app.GetBaseApp().TracingInfo.Start("MidBlock")
 	defer span.End()
 	return app.BaseApp.MidBlock(ctx, height)
 }
 
 func (app *App) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) (res abci.ResponseEndBlock) {
-	_, span := app.tracingInfo.Start("EndBlock")
+	_, span := app.GetBaseApp().TracingInfo.Start("EndBlock")
 	defer span.End()
 	return app.BaseApp.EndBlock(ctx, req)
 }
 
 func (app *App) CheckTx(ctx context.Context, req *abci.RequestCheckTx) (*abci.ResponseCheckTx, error) {
-	_, span := app.tracingInfo.Start("CheckTx")
+	_, span := app.GetBaseApp().TracingInfo.Start("CheckTx")
 	defer span.End()
 	return app.BaseApp.CheckTx(ctx, req)
 }
 
 func (app *App) DeliverTx(ctx sdk.Context, req abci.RequestDeliverTx) abci.ResponseDeliverTx {
 	defer metrics.MeasureDeliverTxDuration(time.Now())
-	_, span := app.tracingInfo.Start("DeliverTx")
+	tracer := app.GetBaseApp().TracingInfo.Tracer
+	spanCtx, span := app.GetBaseApp().TracingInfo.Start("DeliverTx")
 	defer span.End()
+	ctx = ctx.WithSpanAndTracer(spanCtx, tracer)
```

Comment:
qq why do we need to get the tracer here, while in begin/end/mid block we don't need it?